etcdserver/api/membership/cluster_test.go

```diff
@@ -859,3 +880,221 @@ func TestIsReadyToRemoveVotingMember(t *testing.T) {
 		}
 	}
 }
+
+func TestMustDetectDowngrade(t *testing.T) {
+	lv := semver.Must(semver.NewVersion(version.Version))
+	lv = &semver.Version{Major: lv.Major, Minor: lv.Minor}
+	oneMinorHigher := &semver.Version{Major: lv.Major, Minor: lv.Minor + 1}
+	oneMinorLower := &semver.Version{Major: lv.Major, Minor: lv.Minor - 1}
+	downgradeEnabledHigherVersion := &Downgrade{Enabled: true, TargetVersion: oneMinorHigher}
+	downgradeEnabledEqualVersion := &Downgrade{Enabled: true, TargetVersion: lv}
+	downgradeEnabledLowerVersion := &Downgrade{Enabled: true, TargetVersion: oneMinorLower}
+	downgradeDisabled := &Downgrade{Enabled: false}
+
+	tests := []struct {
+		currentVersion *semver.Version
+		downgrade      *Downgrade
+		success        bool
+		message        string
+	}{
+		{
+			nil,
+			downgradeDisabled,
+			true,
+			"",
+		},
+		{
+			oneMinorLower,
+			downgradeDisabled,
+			true,
+			"",
+		},
+		{
+			oneMinorHigher,
+			downgradeDisabled,
+			false,
+			"invalid downgrade; server version is lower than determined cluster version",
```

Comment:
determined cluster version -> downgrade target version?

Reply:
It is should be `determined cluster version`. When downgrade is disabled, the `mustDetectDowngrade` function will compare local version with the determined cluster version.